### PR TITLE
Fixing Extension naming CI Checker

### LINKED ISSUE #4671: Fix CI for extension function name list check
e.g. here https://github.com/duckdb/duckdb/runs/8288060298?check_suite_focus=true

```
File "/home/runner/work/duckdb/duckdb/scripts/generate_extensions_function.py", line 47, in <module>
[20](https://github.com/duckdb/duckdb/runs/8288060298?check_suite_focus=true#step:7:21)
    assert cur_function_map == function_map
[21](https://github.com/duckdb/duckdb/runs/8288060298?check_suite_focus=true#step:7:22)
```

### PR DESCRIPTION
**Pull request by pdet**
Fixes: #4671

### CODE REVIEW

File: scripts/generate_extensions_function.py [hunk 1]
```diff
@@ -25,12 +25,19 @@
 base_functions = {x for x in base_functions}
 
 function_map = {}
+
+import glob
+
+# root_dir needs a trailing slash (i.e. /root/dir/)
+extension_path = {}
+for filename in glob.iglob('/tmp/' + '**/*.duckdb_extension', recursive=True):
+    extension_path[filename.split("/")[-1].split('.')[0]] = filename
```
Comment (Mause) [inline scripts/generate_extensions_function.py:34]: Please use os.path.splitext/basename here instead

File: scripts/generate_extensions_function.py [hunk 2]
```diff
@@ -39,12 +46,16 @@
 if args.validate:
     cur_function_map = {}
     file = open(os.path.join("..","src","include","extension_functions.hpp"),'r')
-    pattern = re.compile("{\"(.*?)\", \"(.*?)\"},")
+    pattern = re.compile("(.*?){\"(.*?)\", \"(.*?)\"},")
```
Comment (Mause) [inline scripts/generate_extensions_function.py:49]: Why not just use findall instead of iterating over every line in the file?

**Comment by pdet** [comment]
@Mytherin if you don't have any comments this is good to go.

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**